repos/fdbt-site/src/pages/defineCapPricingPerDistance.tsx

```diff
@@ -0,0 +1,144 @@
+}: DefineCapPricingPerDistanceProps): ReactElement => {
+    const [numberOfCap, setNumberOfProducts] = useState(numberOfCapInitial);
+    const [capPricingPerDistanceData, setCapPricingPerDistanceData] = useState({
+        [`distanceTo${numberOfCap - 1}`]: 'Max',
+        ...capPricePerDistances,
+    });
+
+    return (
+        <FullColumnLayout title={title} description={description} errors={errors}>
+            <CsrfForm action="/api/capPricingPerDistance" method="post" csrfToken={csrfToken}>
+                <>
+                    <ErrorSummary errors={errors} />
+                    <h1 className="govuk-heading-l" id="cap-pricing-per-distance-heading">
+                        Enter your distance cap details
+                    </h1>
+                    <div className="govuk-grid-row">
+                        <DistanceRow
+                            numberOfCapToDisplay={numberOfCap}
+                            errors={errors}
+                            capPricingPerDistanceData={capPricingPerDistanceData}
+                            setCapPricingPerDistanceData={setCapPricingPerDistanceData}
+                        />
+                        <div className="flex-container">
+                            {numberOfCap < 10 ? (
+                                <button
+                                    id="add-another-button"
+                                    type="button"
+                                    className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3 time-restrictions-button-placement"
+                                    onClick={(): void => {
+                                        setNumberOfProducts(numberOfCap + 1);
+                                        setCapPricingPerDistanceData({
+                                            ...capPricingPerDistanceData,
+                                            [`distanceTo${numberOfCap - 1}`]: '',
+                                            [`distanceTo${numberOfCap}`]: 'Max',
+                                        });
+                                    }}
+                                >
+                                    Add another product
+                                </button>
+                            ) : (
+                                ''
+                            )}
+
+                            {numberOfCap > 1 ? (
+                                <button
+                                    id="remove-button"
+                                    type="button"
+                                    className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3"
+                                    onClick={(): void => {
+                                        setCapPricingPerDistanceData((current) => {
+                                            const copy = { ...current };
+                                            delete copy[`distanceTo${numberOfCap - 1}`];
```

Comment:
we should really be avoiding doing key deletion like this - I think the change in the interfaces will mean this will change anyway.